app/views/Settings.js

```diff
@@ -24,8 +32,22 @@ import Colors from '../constants/colors';
 class SettingsScreen extends Component {
   constructor(props) {
     super(props);
+
+    // Get locales list from i18next for locales menu
+    let localesList = [];
+    for (let key in languages.options.resources) {
```

Comment:
Same thing here. This is copied code from Onboarding1.js

A service or a hook would be great.

```js
const [locale, setLocale, availableLocales] = useLocaleSetByUser();
```